File: samples/server/petstore/scala-play-server/app/api/StoreApiImpl.scala
package api

import model.Order

/**
  * Provides a default implementation for [[StoreApi]].
  */

class StoreApiImpl extends StoreApi {
  /**
    * @inheritdoc
    */
  override def deleteOrder(orderId: String): Unit = {
    // TODO: Implement better logic

    
  }

  /**
    * @inheritdoc
    */
  override def getInventory(): Map[String, Int] = {
    // TODO: Implement better logic

    Map.empty[String, Int]
  }

  /**
    * @inheritdoc
    */
  override def getOrderById(orderId: Long): Order = {
    // TODO: Implement better logic

    Order(None, None, None, None, None, None)
  }

  /**
    * @inheritdoc
    */
  override def placeOrder(order: Order): Order = {
    // TODO: Implement better logic

    Order(None, None, None, None, None, None)
  }
}

File: samples/server/petstore/kotlin-misk/src/main/kotlin/org/openapitools/server/api/api/StoreApiImpl.kt
package org.openapitools.server.api.api

import jakarta.inject.Inject
import jakarta.inject.Singleton
import jakarta.validation.Valid
import jakarta.validation.constraints.DecimalMax
import jakarta.validation.constraints.DecimalMin
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size
import misk.web.HttpCall
import misk.web.PathParam
import misk.web.QueryParam
import misk.web.RequestBody
import misk.web.RequestHeader
import org.openapitools.server.api.model.Order

/**
 * @TODO("Fill out implementation")
 */
@Singleton
class StoreApiImpl @Inject constructor(
): StoreApi {

    override fun deleteOrder(@PathParam("orderId") orderId: kotlin.String) {
        TODO()
    }

    override fun getInventory(): kotlin.collections.Map<kotlin.String, kotlin.Int> {
        TODO()
    }

    override fun getOrderById(@Min(1L) @Max(5L) @PathParam("orderId") orderId: kotlin.Long): Order {
        TODO()
    }

    override fun placeOrder(@Valid @RequestBody order: Order): Order {
        TODO()
    }
}
